File: service/jvm-java-quarkus/src/main/java/com/ecobenchmark/controllers/getlists/TaskResponse.java
package com.ecobenchmark.controllers.getlists;

import io.quarkus.runtime.annotations.RegisterForReflection;

import java.time.Instant;
import java.util.UUID;

@RegisterForReflection
public class TaskResponse {

    private UUID id;

    private String name;

    private String description;

    private Instant creationDate;

    public TaskResponse(UUID id, String name, String description, Instant creationDate) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.creationDate = creationDate;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Instant getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Instant creationDate) {
        this.creationDate = creationDate;
    }
}

File: service/jvm-kotlin-spring/src/main/kotlin/com/ecobenchmark/kotlinspringjpa/controllers/addtask/TaskResponse.kt
package com.ecobenchmark.kotlinspringjpa.controllers.addtask

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant
import java.util.UUID

data class TaskResponse(
    val id: UUID,
    @JsonProperty("list_id")
    val listId: UUID,
    val name: String,
    val description: String,
    @JsonProperty("creation_date")
    val creationDate: Instant
)